packages/esm-patient-list-management-app/src/hooks/usePatientListing.tsx

```diff
@@ -232,52 +244,159 @@ export const usePatientListing = () => {
+        setLoading(false);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  const getIIT = async ({ currentPage, pageSize }) => {
+    try{
+      if (currentPage === 0) {
+        setTableData([]);
+        setLoading(true)
+      };
+
+      const url = `/ws/rest/v1/ssemr/dashboard/interruptedInTreatment?startDate=${startDate}&endDate=${endDate}&page=${currentPage}&size=${pageSize}`;
+
+      setCurrentPaginationState((prev) => ({ ...prev, done: false }));
+
+      const { data } = await openmrsFetch(url);
+
+      if (data?.results?.length > 0)
+        setTableData((prev) => [...prev, ...data.results]);
+
+      if (data?.results?.length === pageSize)
+        setCurrentPaginationState((prev) => ({
+          ...prev,
+          page: ++prev.page,
+        }))
+
+      if (data?.results?.length === 0 || data?.results?.length < pageSize) {
+        setCurrentPaginationState((prev) => ({ ...prev, page: 0, done: true }));
+        setLoading(false);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  const getTransferredOut = async ({ currentPage, pageSize }) => {
+    try{
+      if (currentPage === 0) {
+        setTableData([]);
+        setLoading(true)
+      };
+
+      const url = `/ws/rest/v1/ssemr/dashboard/transferredOut?startDate=${startDate}&endDate=${endDate}&page=${currentPage}&size=${pageSize}`;
+
+      setCurrentPaginationState((prev) => ({ ...prev, done: false }));
+
+      const { data } = await openmrsFetch(url);
+
+      if (data?.results?.length > 0)
+        setTableData((prev) => [...prev, ...data.results]);
+
+      if (data?.results?.length === pageSize)
+        setCurrentPaginationState((prev) => ({
+          ...prev,
+          page: ++prev.page,
+        }))
+
+      if (data?.results?.length === 0 || data?.results?.length < pageSize) {
+        setCurrentPaginationState((prev) => ({ ...prev, page: 0, done: true }));
+        setLoading(false);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  const getDeceased = async ({ currentPage, pageSize }) => {
+    try{
+      if (currentPage === 0) {
+        setTableData([]);
+        setLoading(true)
+      };
+
+      const url = `/ws/rest/v1/ssemr/dashboard/deceased?startDate=${startDate}&endDate=${endDate}&page=${currentPage}&size=${pageSize}`;
+
+      setCurrentPaginationState((prev) => ({ ...prev, done: false }));
+
+      const { data } = await openmrsFetch(url);
+
+      if (data?.results?.length > 0)
+        setTableData((prev) => [...prev, ...data.results]);
+
+      if (data?.results?.length === pageSize)
+        setCurrentPaginationState((prev) => ({
+          ...prev,
+          page: ++prev.page,
+        }))
+
+      if (data?.results?.length === 0 || data?.results?.length < pageSize) {
+        setCurrentPaginationState((prev) => ({ ...prev, page: 0, done: true }));
+        setLoading(false);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
```

Comment:
@Samstar10 I think we can optimize this further by introducing a reusable component since they are all performing the same task.

cc: @PiusKariuki what do you think?

Reply:
I agree. @Michaelndula